docs/syntax/syntax.md

```diff
@@ -205,95 +205,429 @@ Is below, but it won't be parsed into the document.
+```
+````
+:::
+:::{grid-item-card}
+```{math}
+:label: eq:ref
+a = 1
+```
+:::
+
+::::
+
+To refer to the number or title/caption of a numbered object, you must first enable placeholder replacements.
+
+```python
+myst_link_placeholders = True
 ```
 
-but can also accept "nested" syntax (like bold text) and will recognise document paths that include extensions (e.g. `syntax/syntax` or `syntax/syntax.md`)
+You can then use the `{number}` and `{name}` placeholders in the link text:
+
+|                   Examples                    |                                             |
+| :-------------------------------------------- | :------------------------------------------ |
+| `[section {number} *"{name}"*](#section:ref)` | [section {number} *"{name}"*](#section:ref) |
+| `[figure {number} *"{name}"*](#figure:ref)`   | [figure {number} *"{name}"*](#figure:ref)   |
+| `[code {number} *"{name}"*](#code:ref)`       | [code {number} *"{name}"*](#code:ref)       |
+| `[table {number} *"{name}"*](#table:ref)`     | [table {number} *"{name}"*](#table:ref)     |
+| `[equation {number}](#eq:ref)`                | [equation {number}](#eq:ref)                |
+
+
+(syntax/referencing/myst-inv)=
+### Cross-project (inventory) targets
 
-Using the same example, see this ref: [](syntax/targets), here is a reference back to the top of
-this page: [my text with **nested** $\alpha$ syntax](syntax/core), and here is a reference to another page (`[](../intro.md)`): [](../intro.md).
+For HTML builds, the inventory that is generated for the project, is also written to a file called `objects.inv`,
+and contains the "endpoints" for all the targets in the project, which can be combined with the base URL of the project to create a full URL to the target.
 
-```{note}
-If you wish to have the target's title inserted into your text, you can
-leave the "text" section of the markdown link empty. For example, this
-markdown: `[](syntax.md)` will result in: [](syntax.md).
+Any HTML site then built with Sphinx should contain this file at the root of the site.
+You can export this file to a readable YAML, using the `myst-inv` command line tool, pointing to either a local file or remote URL:
+
+```yaml
+# $ myst-inv -d "py" -o "module" -n "datetime"  https://docs.python.org/3.7/objects.inv
+name: Python
+version: '3.7'
+objects:
+  py:
+    module:
+      datetime:
+        loc: library/datetime.html#module-datetime
+        disp: '-'
 ```
 
+To load external inventories into your Sphinx project, you must load the [`sphinx.ext.intersphinx` extension](myst:sphinx#usage/extensions/intersphinx), and set the `intersphinx_mapping` configuration option, e.g.:
+
+```python
+extensions = ["myst_parser", "sphinx.ext.intersphinx"]
+intersphinx_mapping = {
+    "python": ("https://docs.python.org/3.7", None),
+}
+```
+
+You can then use `myst:key#target` to reference targets in an external inventory, in a similar fashion to the [project-wide targets](#syntax/referencing/myst-project).
```

Comment:
Crazy idea, what if we re-used the `project` (or renamed to `id`) scheme here, and let people give an external project as a query parameter? e.g. `<project:?name=my-proj#mytarget>` then it'd be one fewer schema people would have to remember.

Reply:
I'm actually changing how the text in the query works, because I felt it was a bit verbose to specify the domain/object type.

Instead of:

```
[](project:?d=py&o=class#ref)
[](myst:sphinx?d=py&o=class#ref)
```

I was moving to (note you can also now drop the scheme if the URL start with `?`):

```
[](?py:class#ref)
[](myst:sphinx?py:class#ref)
```

This tracks closer with, the current "role" way of doing it:

```
{py:class}`ref`
```

so the query string no longer maps to the "semi-standard" (its not actually in the spec) `key1=val&key2=val` format, but the trade-off is that its easier to write 🤷 

I've also replaced the pattern matching `?pat#*target` with just recognising that if the target starts with a `*` then it means match the end, e.g.

```
(areallylonglonglongtarget)=
# title
[](#*target)
```

Reply:
do you think that makes sense?

Reply:
I think that staying within standard links is a positive. This means, for example you can use standard URL parsing libraries, that all support query parameters, for example in Javascript:

```typescript
const u = new URL('myst:sphinx?d=py&o=class#ref')
u.protocol
u.hash
u.searchParams.get('o')
```

These are things that are likely supported in any major language, which is a nice thing to not have to reimagine and parse.

Can we just change the d/o --> a type to make it less verbose?

Is it very often that we will have to even use this syntax, because the default target will just go through til it finds the reference, so it is only when that fails that you have to become more specific?

I am coming around to @choldgraf 's idea of merging the two syntaxes into a `project:` protocol.

Reply:
> Can we just change the d/o --> a type to make it less verbose?

I'm not quite sure what you mean by "a type" here?

> Is it very often that we will have to even use this syntax

In the sphinx world, I would say yes, it will be widely used

Reply:
Its just err, this looks horrible to me `<project:?name=my-proj&d=py&o=class#mytarget>`, and not very readable, whereas `<project:my-proj?py:class#mytarget>` feels so much more concise, and easy to follow?

Reply:
> Can we just change the d/o --> a type to make it less verbose?

By that I mean: `'myst:jupyterbook?t=py:class#ref'` where `t` is named something sensible. In your example I guess you are saying that the `t=` may even be removed?

I agree that putting too much in query parameters makes it pretty unreadable, I am unsure though how much we would actually use the `?t=py:class` when the implementation by default looks through *all* ids regardless of domain, which is different than sphinx, so maybe it won't be as necessary, and optimizing for terse-ness isn't as necessary.

Reply:
> by default looks through all ids regardless of domain, which is different than sphinx

there is an `{any}` role which does this, but most people choose(?) to use a specific role like `{ref}`

Reply:
I guess, if were to still use this, I would go for `?f=py:class`, since its a filter; you can also do `?f=py:*` and `?f=*:class`